src/Swarm/TUI/Controller.hs

```diff
@@ -164,9 +164,31 @@ runFrame = do
         | lgTPS >= 0 = oneSecond `div` (1 `shiftL` lgTPS)
         | otherwise  = oneSecond * (1 `shiftL` abs lgTPS)
 
+  -- Update TPS/FPS counters every second
+  infoUpdateTime <- use (uiState . lastInfoTime)
+  let updateTime = toNanoSecs $ diffTimeSpec curTime infoUpdateTime
+  when (updateTime >= oneSecond) $ do
+    -- Wait for at least one second to have elapsed
+    when (infoUpdateTime /= 0) $ do
+      -- calc is an helper function to compute a (tick|frame) per second value
+      let calc counterLens valueLens = do
+            count <- use (uiState . counterLens)
+            let countSec = toInteger count * oneSecond
+            uiState . valueLens .= fromIntegral countSec / fromIntegral updateTime
+      calc frameTicks uiTPS
+      calc frameCount uiFPS
+      -- ensure this frame gets drawn
+      gameState . needsRedraw .= True
+    -- Reset the counter and wait another seconds for the next update
+    uiState . frameTicks .= 0
```

Comment:
Let's change the name of `frameTicks` to `tickCount`, since it no longer counts the ticks in a single frame, but the ticks since the last info widget update.